main.tf

```diff
@@ -20,12 +22,14 @@ module "spacelift_environment" {
 
   for_each = toset(var.stack_config_files)
 
-  trigger_policy_id = spacelift_policy.trigger_global.id
+  trigger_policy_id = join("", spacelift_policy.trigger_global.*.id)
   push_policy_id    = spacelift_policy.push.id
   plan_policy_id    = spacelift_policy.plan.id
   stack_config_name = trimsuffix(each.key, ".yaml")
   components        = try(module.yaml_stack_config[each.key].config.0.components.terraform, {})
+  imports           = [for import in try(module.yaml_stack_config[each.key].config.0.imports, []) : format("%s/%s.yaml", local.stack_config_path, import)]
```

Comment:
I don't like the `try` blocks either, but they make the code more readable.
And taking into account Terraform "strict" type system, we can't just use `lookup` with a default value (since Terraform wants the items of a collection to have the same type and count:

```
 result = lookup(var.input, "imports", [])

Invalid value for "default" parameter: the default value must have the same
type as the map elements.
```

This "ugly" thing works

```
  imports           = [for import in (lookup(module.yaml_stack_config[each.key].config[0], "imports", null) != null ? lookup(module.yaml_stack_config[each.key].config[0], "imports") : []) : format("%s/%s.yaml", local.stack_config_path, import)]
```

But I think it's much worse than using `try`

Reply:
I agree that using the ternary is ugly, but 

1. It looks to me like you are [guaranteed to have an `imports` key](https://github.com/cloudposse/terraform-yaml-stack-config/blob/83d40f882e7f7a4047fbd4322eedddb4bc3e780f/main.tf#L9-L26) 
2. You could use `coalesce` which I like better

```suggestion
  components        = coalesce(module.yaml_stack_config[each.key].config.0.components.terraform, {})
  imports           = [for import in coalesce(lookup(module.yaml_stack_config[each.key].config.0, "imports", null), []) : format("%s/%s.yaml", local.stack_config_path, import)]
```

Reply:
this 

```
imports           = [for import in coalesce(lookup(module.yaml_stack_config[each.key].config.0, "imports", null), []) : format("%s/%s.yaml", local.stack_config_path, import)]
```

works ok - `coalesce` works with a list as default.

But this

```
coalesce(module.yaml_stack_config[each.key].config.0.components.terraform, {})
```

is not working:

```
This default value is not compatible with the variable's type constraint: all
map elements must have the same type.
```

We can remove `try` from one line, but not from the other.

@Nuru  @osterman  I'd leave it for now, too much time to figure it our correctly